pkg/modulereader/resreader.go

```diff
@@ -39,16 +41,87 @@ type VarInfo struct {
 	Required    bool
 }
 
+// OutputInfo stores information about module output values
+type OutputInfo struct {
+	Name        string
+	Description string `yaml:",omitempty"`
+	Sensitive   bool   `yaml:",omitempty"`
+	// DependsOn   []string `yaml:"depends_on,omitempty"`
+}
+
+// UnmarshalYAML supports parsing YAML OutputInfo fields as a simple list of
+// strings or as a list of maps directly into OutputInfo struct
+func (mo *OutputInfo) UnmarshalYAML(value *yaml.Node) error {
+	var name string
+
+	switch value.Kind {
+	case yaml.ScalarNode:
+		err := value.Decode(&name)
+		if err == nil {
+			mo.Name = name
+			return nil
+		}
+		return fmt.Errorf("line %d: %s", value.Line, err)
+	case yaml.MappingNode:
+		var raw map[string]interface{}
+		if err := value.Decode((*map[string]interface{})(&raw)); err != nil {
+			return fmt.Errorf("line %d: %s", value.Line, err)
+		}
+		err := enforceMapKeys(raw, []string{"name"}, []string{"description", "sensitive"})
+		if err != nil {
+			return fmt.Errorf("line %d: %s", value.Line, err)
+		}
+		type rawOutputInfo OutputInfo
+		if err := value.Decode((*rawOutputInfo)(mo)); err != nil {
+			return fmt.Errorf("line %d: %s", value.Line, err)
+		}
+		return nil
+	default:
+		return fmt.Errorf("line %d: outputs must each be a string or a map{name: string, description: string, sensitive: bool}", value.Line)
+	}
+}
+
+// enforceMapKeys ensures the presence of required keys and absence of unallowed
+// keys with a useful error message
+func enforceMapKeys(input map[string]interface{}, requiredKeys []string, optionalKeys []string) error {
```

Comment:
nit. It can be written less verbosly : 
```
satisfied = map[string]bool {"name": false, "description": true, "sensitive": true}
for _, key := range fields {
  if _, ok := satisfied[key]; !ok {
     return fmt.Errorf("provided invalid key: %#v", key)
  }
  satisfied[key] = true
}
for key, sat := range satisfied {
  if !sat {
    return fmt.Errorf("missing required key: %#v", key)
  }
}
```

Reply:
A similar solution has been adopted on the head of PR. Additionally prevents accidental inclusion of the same key as required and optional.